.github/workflows/integration-in-memory-tests.yml

```diff
@@ -0,0 +1,145 @@
+#
+# Workflow to run in-memory integration tests
+# Test matrix is defined in .github/integration-in-memory-tests.yml
+#
+name: Integration In-Memory Tests
+run-name: Integration In-Memory Tests
+on:
+  merge_group:
+  pull_request:
+  push:
+    tags:
+      - "*"
+  workflow_dispatch:
+    inputs:
+      cl_ref:
+        description: 'The ref to checkout, defaults to the calling branch'
+        required: false
+        type: string
+
+# Only run 1 of this workflow at a time per PR
+concurrency:
+  group: ${{ github.ref }}-${{ github.repository }}-${{ github.event_name }}--integration-tests
+  cancel-in-progress: true
+
+jobs:
+  changes:
+    environment: integration
+    name: Check Paths That Require Tests To Run
+    runs-on: ubuntu-latest
+    # We don't directly merge dependabot PRs, so let's not waste the resources
+    if: github.actor != 'dependabot[bot]'
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v4.2.1
+        with:
+          repository: smartcontractkit/chainlink
+          ref: ${{ inputs.cl_ref }}
+      - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: changes
+        with:
+          filters: |
+            github_ci_changes:
+              - '.github/workflows/integration-tests.yml'
+              - '.github/workflows/integration-in-memory-tests.yml'
+              - '.github/integration-in-memory-tests.yml'
+            core_changes:
+              - '**/*.go'
+              - '**/*go.sum'
+              - '**/*go.mod'
+              - '**/*Dockerfile'
+              - 'core/**/migrations/*.sql'
+              - 'core/**/config/**/*.toml'
+              - 'integration-tests/**/*.toml'
+            ccip_changes:
+              - '**/*ccip*'
+              - '**/*ccip*/**'
+      - name: Ignore Filter On Workflow Dispatch
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        id: ignore-filter
+        run: echo "changes=true" >> $GITHUB_OUTPUT
+    outputs:
+      github_ci_changes: ${{ steps.ignore-filter.outputs.changes || steps.changes.outputs.github_ci_changes }}
+      core_changes: ${{ steps.ignore-filter.outputs.changes || steps.changes.outputs.core_changes }}
+      ccip_changes: ${{ steps.ignore-filter.outputs.changes || steps.changes.outputs.ccip_changes }}
```

Comment:
curious, why do you need 3 outputs, if later they are not really used? why not collapse them all into one and use that later?

meaning:
```
outputs:
      changes: ${{ steps.ignore-filter.outputs.changes || steps.changes.outputs.github_ci_changes || steps.changes.outputs.core_changes  || steps.changes.outputs.ccip_changes}}
```

and then later:
```
 if: github.event_name == 'pull_request' && ( needs.changes.chagnes == 'true)
```